docs/features/skiplocalsinit.md

```diff
@@ -0,0 +1,24 @@
+
+# Feature design for SkipLocalsInit
+
+The feature is to add a new well-known attribute to the compiler, System.Runtime.CompilerServices.SkipLocalsInit that causes method bodies nested "inside" the scope of the attribute to elide the ".locals init" CIL directive that causes the CLR to zero-init local variables and space reserved using the "localloc" instruction. "Nested inside the scope of the attribute" is a concept defined as follows:
+
+1. When the attribute is applied to a module, all emitted methods inside that module, including generated methods, will skip local initialization.
+
+2. When the attribute is applied to a type (including interfaces), all methods inside that type, including generated methods, will skip local initialization.
```

Comment:
Does that include nested types defined inside the type?

Reply:
Yup, I'll add a mention. See test SkipLocalsInitAttributeOnClassKeepsPropagatingToNestedClasses